block_course_contacts.php

```diff
@@ -199,7 +199,11 @@ public function get_content() {
         $userfields = user_picture::fields('u', ['lastaccess', 'phone1', 'description']);
 
         $currentgroup = groups_get_course_group($COURSE, true, false);
-        $content .= groups_print_course_menu($COURSE, new moodle_url('/course/view.php?id='.$courseid), true);
+        if ($this->config->group) {
+            $url = $PAGE->url;
```

Comment:
Same as above:

You should probably be using $this->page instead of the global $PAGE variable.

> global $PAGE cannot be used in block classes. Use $this->page.